### PR TITLE
feat(frontend): migrate insert to region server

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

* Introduce `RegionRequestHandler` to proxy Region Requests to Region Server(distributed mode) or invoking the handle method(standalone mode)
* Provide a converter that transforms an InsertRequest in db proto into an InsertRequest in region proto
* Provide a converter that transforms an InsertRequest in table proto into an InsertRequest in region proto
* Rewrite splitter to be split Region Insert Request, not Table Insert Request.
* Mak Statement Executor in frontend, `Inserter` in frontend, Statement Executor in dist instance, and `DistInserter` handle insert through region request handler.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/issues/2065

### CODE REVIEW

File: src/frontend/src/instance/distributed.rs [hunk 1]
```diff
@@ -273,19 +272,14 @@ impl DistInstance {
                 self.drop_table(table_name).await
             }
             Statement::Insert(insert) => {
-                let (catalog, schema, _) =
-                    table_idents_to_full_name(insert.table_name(), query_ctx.clone())
-                        .map_err(BoxedError::new)
-                        .context(error::ExternalSnafu)?;
-
                 let insert_request =
                     SqlHandler::insert_to_request(self.catalog_manager.clone(), &insert, query_ctx)
```
Comment (fengjiachun) [inline src/frontend/src/instance/distributed.rs:275]: Maybe next PR we can directly convert `insert`into row format.

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/frontend/src/instance/distributed.rs (hunk 1 above)
Comment (zhongzc) [reply src/frontend/src/instance/distributed.rs:275]: Start immediately

File: src/frontend/src/statement/copy_table_from.rs [hunk 2]
```diff
@@ -322,14 +327,17 @@ impl StatementExecutor {
...
-                    //TODO: support multi-regions
-                    region_number: 0,
-                }));
+                pending.push(self.handle_table_insert_request(
+                    InsertRequest {
+                        catalog_name: req.catalog_name.to_string(),
+                        schema_name: req.schema_name.to_string(),
+                        table_name: req.table_name.to_string(),
+                        columns_values,
+                        //TODO: support multi-regions
```
Comment (waynexia) [inline src/frontend/src/statement/copy_table_from.rs:336]: nit:
```suggestion
                        // TODO: support multi-regions
```

File: src/frontend/src/instance/distributed/inserter.rs [hunk 3]
```diff
@@ -13,180 +13,134 @@
...
+            })?;
 
-        self.request_datanodes(inserts).await
+        let table_info = table.table_info();
+        let request = Inserter::convert_req_table_to_region(&table_info, request)?;
+        self.insert_region_requests(request).await
     }
 
-    /// Splits multiple table [InsertRequest]s into multiple GRPC [InsertRequests]s, each of which
+    /// Splits GRPC [InsertRequests] into multiple GRPC [InsertRequests]s, each of which
```
Comment (waynexia) [inline src/frontend/src/instance/distributed/inserter.rs:89]: nit:
```suggestion
    /// Splits gRPC [InsertRequests] into multiple gRPC [InsertRequests]s, each of which
```

File: src/frontend/src/instance/distributed.rs [hunk 4]
```diff
@@ -628,6 +591,44 @@ impl GrpcQueryHandler for DistInstance {
...
+        Arc::new(Self { catalog_manager })
+    }
+}
+
+#[async_trait]
+impl RegionRequestHandler for DistRegionRequestHandler {
+    async fn handle(
+        &self,
+        request: region_request::Body,
+        _ctx: QueryContextRef,
```
Comment (waynexia) [inline src/frontend/src/instance/distributed.rs:609]: Dist request should pass `trace_id` from this context.

File: src/frontend/src/instance/distributed.rs [hunk 5]
```diff
@@ -628,6 +591,44 @@ impl GrpcQueryHandler for DistInstance {
...
+                })
+            }
+            region_request::Body::Deletes(_)
+            | region_request::Body::Create(_)
+            | region_request::Body::Drop(_)
+            | region_request::Body::Open(_)
+            | region_request::Body::Close(_)
+            | region_request::Body::Alter(_)
+            | region_request::Body::Flush(_)
+            | region_request::Body::Compact(_) => todo!(),
```
Comment (waynexia) [inline src/frontend/src/instance/distributed.rs:627]: prefer to return an "unimplemented" error instead of panic

**Review by waynexia — APPROVED** [review]
Mostly LGTM

**Pull request merged**